<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Mirko Windhoff</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>795</width>
    <height>736</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtLobby</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <widget class="InputLine" name="inputLineEdit"/>
    </item>
    <item row="0" column="0">
     <widget class="QStackedWidget" name="lobbyStackedWidget">
      <property name="contextMenuPolicy">
       <enum>Qt::CustomContextMenu</enum>
      </property>
      <widget class="QWidget" name="lobbyStackedWidgetPage6"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>795</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Program">
    <property name="title">
     <string>&amp;Program</string>
    </property>
    <addaction name="action_Connect"/>
    <addaction name="action_Disconnect"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_About"/>
    <addaction name="actionHelp"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionReload_maps_and_mods"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
    <addaction name="actionStylesheet"/>
    <addaction name="actionScripting"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Battle_list"/>
    <addaction name="actionBattle_Info"/>
    <addaction name="action_User_list"/>
    <addaction name="actionMap_list"/>
    <addaction name="actionGroups"/>
    <addaction name="actionDownloads"/>
   </widget>
   <widget class="QMenu" name="menu_Game">
    <property name="title">
     <string>&amp;Game</string>
    </property>
    <addaction name="actionStart_Spring"/>
    <addaction name="action_Spring_Settings"/>
    <addaction name="action_Host_a_Battle"/>
   </widget>
   <addaction name="menu_Program"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Game"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="userListDockWidget">
   <property name="font">
    <font>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="focusPolicy">
    <enum>Qt::TabFocus</enum>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>User List</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout">
         <item row="0" column="0">
          <widget class="QToolButton" name="userListClearLineEditButton">
           <property name="toolTip">
            <string>Clear the userlistfilter string</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="arrowType">
            <enum>Qt::RightArrow</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="userListLineEdit">
           <property name="toolTip">
            <string>This userlist filter string is used to match playernames</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="Users" name="users">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideNone</enum>
         </property>
         <property name="indentation">
          <number>0</number>
         </property>
         <property name="rootIsDecorated">
          <bool>false</bool>
         </property>
         <property name="itemsExpandable">
          <bool>false</bool>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <property name="expandsOnDoubleClick">
          <bool>false</bool>
         </property>
         <attribute name="headerStretchLastSection">
          <bool>true</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="usersInCurrentChannel">
           <property name="toolTip">
            <string># Users</string>
           </property>
           <property name="text">
            <string>Users: 0</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="availableSlots">
           <property name="toolTip">
            <string># Users</string>
           </property>
           <property name="text">
            <string>available Slots: 0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="battleListDockWidget">
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>Battle List</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0" colspan="6">
      <widget class="Battles" name="battles">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="indentation">
        <number>0</number>
       </property>
       <property name="rootIsDecorated">
        <bool>false</bool>
       </property>
       <property name="itemsExpandable">
        <bool>false</bool>
       </property>
       <property name="sortingEnabled">
        <bool>true</bool>
       </property>
       <property name="expandsOnDoubleClick">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QToolButton" name="battleListClearLineEditButton">
       <property name="toolTip">
        <string>Clear the battlefilter string</string>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="arrowType">
        <enum>Qt::RightArrow</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="battleListLineEdit">
       <property name="toolTip">
        <string>This filter string is used to match battlenames</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Enter a filter word here.</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>61</width>
         <height>31</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="5">
      <widget class="QPushButton" name="hostPushButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>&amp;Host</string>
       </property>
      </widget>
     </item>
     <item row="1" column="4">
      <widget class="QPushButton" name="joinPushButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>&amp;Join</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="battleInfoDockWidget">
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>Battle Info</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="UsersTreeView" name="battleInfoTreeView">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="indentation">
        <number>0</number>
       </property>
       <property name="rootIsDecorated">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="tabsDockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Tabs</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="0" column="0">
      <widget class="ToolBarWidget" name="toolBarWidget" native="true"/>
     </item>
     <item row="0" column="1">
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>744</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="0" colspan="2">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>748</width>
         <height>2</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_Exit">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/Exit.png</normaloff>:/icons/Exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Connect">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/Connect_creating.png</normaloff>:/icons/Connect_creating.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Login</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+L</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionReload_maps_and_mods">
   <property name="text">
    <string>&amp;Reload maps and mods</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_Disconnect">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/Connect_no.png</normaloff>:/icons/Connect_no.png</iconset>
   </property>
   <property name="text">
    <string>Logou&amp;t</string>
   </property>
  </action>
  <action name="action_Battle_list">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/battle_list.xpm</normaloff>:/icons/battle_list.xpm</iconset>
   </property>
   <property name="text">
    <string>&amp;Battle list</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+B</string>
   </property>
  </action>
  <action name="action_User_list">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/userlist.png</normaloff>:/icons/userlist.png</iconset>
   </property>
   <property name="text">
    <string>&amp;User list</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+U</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="action_Join">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Join</string>
   </property>
  </action>
  <action name="actionLeave">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/closetab.png</normaloff>:/icons/closetab.png</iconset>
   </property>
   <property name="text">
    <string>Leave</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionStart_Spring">
   <property name="text">
    <string>Try to &amp;reconnect</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>&amp;Preferences</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+P</string>
   </property>
  </action>
  <action name="action_Spring_Settings">
   <property name="text">
    <string>Spring Settin&amp;gs</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionMap_list">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Map list</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionStylesheet">
   <property name="text">
    <string>&amp;Stylesheet</string>
   </property>
  </action>
  <action name="actionGroups">
   <property name="text">
    <string>&amp;Groups</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+G</string>
   </property>
  </action>
  <action name="actionScripting">
   <property name="text">
    <string>S&amp;cripting</string>
   </property>
  </action>
  <action name="actionBattle_Info">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/rank6.xpm</normaloff>:/icons/rank6.xpm</iconset>
   </property>
   <property name="text">
    <string>Battle &amp;Info</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+I</string>
   </property>
  </action>
  <action name="actionDownloads">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/download_map.xpm</normaloff>:/icons/download_map.xpm</iconset>
   </property>
   <property name="text">
    <string>&amp;Downloads</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+D</string>
   </property>
  </action>
  <action name="action_Host_a_Battle">
   <property name="text">
    <string>&amp;Host a Battle</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+H</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="4" margin="4"/>
 <customwidgets>
  <customwidget>
   <class>InputLine</class>
   <extends>QLineEdit</extends>
   <header>InputLine.h</header>
   <slots>
    <slot>onTabChanged(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Users</class>
   <extends>QTreeView</extends>
   <header>Users.h</header>
  </customwidget>
  <customwidget>
   <class>Battles</class>
   <extends>QTreeView</extends>
   <header>Battles.h</header>
  </customwidget>
  <customwidget>
   <class>UsersTreeView</class>
   <extends>QTreeView</extends>
   <header>UsersTreeView.h</header>
  </customwidget>
  <customwidget>
   <class>ToolBarWidget</class>
   <extends>QWidget</extends>
   <header>ToolBarWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>action_Exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>userListClearLineEditButton</sender>
   <signal>clicked()</signal>
   <receiver>userListLineEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>732</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>864</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>battleListClearLineEditButton</sender>
   <signal>clicked()</signal>
   <receiver>battleListLineEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>31</x>
     <y>675</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>674</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStylesheet</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showStylesheetEditor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Battle_list</sender>
   <signal>toggled(bool)</signal>
   <receiver>battleListDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>602</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBattle_Info</sender>
   <signal>triggered(bool)</signal>
   <receiver>battleInfoDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>854</x>
     <y>573</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>battleListDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>action_Battle_list</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>573</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_User_list</sender>
   <signal>toggled(bool)</signal>
   <receiver>userListDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>833</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>userListDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>action_User_list</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>833</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGroups</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showGroupsDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>battleInfoDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionBattle_Info</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>651</x>
     <y>582</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
 <slots>
  <slot>showStylesheetEditor()</slot>
  <slot>showGroupsDialog()</slot>
 </slots>
</ui>
